src/extension.ts

```diff
@@ -38,8 +40,21 @@ export function activate(context: vscode.ExtensionContext): void {
 	context.subscriptions.push(openWithCommand);
 	context.subscriptions.push(telemetryReporter);
 	context.subscriptions.push(HexEditorProvider.register(context, telemetryReporter, dataInspectorProvider));
+	myStatusBarItem = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Right, 100);
+	context.subscriptions.push(myStatusBarItem);
+}
+
+function updateStatusBarItem(count: number): void {
+	if (count > 0) {
+		myStatusBarItem.text = `${count} byte(s) selected`;
+		myStatusBarItem.show();
+	} else {
+		myStatusBarItem.hide();
+	}
 }
 
+HexEditorProvider.onDidChangeSelectionCount((e: number) => updateStatusBarItem(e));
```

Comment:
Organizational thing: I would consider wrapping up the status bar entry in a small class, and then injecting that when creating a new HexEditorProvider. The HexEditorProvider could then call some `update()` method on it.

Also, we should hide the badge when the hex editor is no longer the active editor, and show it when it is.

Reply:
agreed regarding the class thing. not 100% sure how to hide the status bar depending on the active editor. I see there's a `vscode.window.onDidChangeActiveTextEditor(` but I'm not sure how to check if this active text editor is a hexeditor or not. Do you happen to know a quick/easy way to do this?

Reply:
Not off hand, I would have to experiment myself honestly 😅

Reply:
We used to set a context key when the editor opened to reveal the sidebar. This has since been refactored and was a bit glitchy. The API needed to know if the currently open editor is the hex editor is https://github.com/microsoft/vscode/issues/133532 and should be finalized next milestone.  There may be another built in context key that allows for like the current open webview. I remember testing something like that but don't remember what it is off the top of my head.

Reply:
so this wouldnt be able to be merged without that change in the API?

Reply:
We could allow the hex editor to use the proposed API. Will let us get some dogfooding and be able to get this PR in 🙂  (though we would likely have to gate it to VS Code Insiders builds until the API stabilizes)

Reply:
would it make sense to ship a version with both the original solution with the selected byte count in the main UI below the various UTF versions of the data AND the status bar feature, but have the status bar only available for those insider builds?

Reply:
I wouldn't want to rely on proposed API within the hex editor. I did find the context key which dictates the currently active custom editor, that can probably be used. Here's the source code from VS Code https://github.com/microsoft/vscode/blob/4a130c40ed876644ed8af2943809d08221375408/src/vs/workbench/contrib/customEditor/common/customEditor.ts#L18-L21

Reply:
The tabs API needed for this will be finalized this iteration so you should be able to use that next week as well to wrap us this PR.

Reply:
Awesome, thanks for the heads up! 

It might be a while before I get a chance to work on this again, so just a heads up in case this is a priority for the team.